src/Microsoft.AspNet.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -103,12 +103,37 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
                     validationParameters.IssuerSigningKeys = (validationParameters.IssuerSigningKeys == null ? _configuration.SigningKeys : validationParameters.IssuerSigningKeys.Concat(_configuration.SigningKeys));
                 }
 
+                List<Exception> validationFailures = null;
                 SecurityToken validatedToken;
                 foreach (var validator in Options.SecurityTokenValidators)
                 {
                     if (validator.CanReadToken(token))
                     {
-                        var principal = validator.ValidateToken(token, validationParameters, out validatedToken);
+                        ClaimsPrincipal principal;
+                        try
+                        {
+                            principal = validator.ValidateToken(token, validationParameters, out validatedToken);
+                        }
+                        catch (Exception ex)
+                        {
+                            Logger.LogInformation("Failed to validate the token: " + token, ex);
+
+                            // Refresh the configuration for exceptions that may be caused by key rollovers. The user can also request a refresh in the event.
+                            if (Options.RefreshOnIssuerKeyNotFound && ex.GetType().Equals(typeof(SecurityTokenSignatureKeyNotFoundException)))
```

Comment:
For multi IDP's, it may be the case that the token should not have been processed. Refreshing here doesn't seem optimal.